ui/src/components/LocationFieldset.js

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, {Component} from 'react';
+import ReactDOM from 'react-dom';
+import {FormattedHTMLMessage, defineMessages, injectIntl} from 'react-intl';
+import {withRouter} from 'react-router-dom';
+import Button from '@material/react-button';
+import TextField, {HelperText, Input} from '@material/react-text-field';
+import Utils from './../utils/Utils';
+
+const MESSAGES = defineMessages({
+  hideMap: {
+    id: 'LocationFieldset.hideMap',
+    defaultMessage: 'Hide map',
+    description: 'Label on a button to hide a map display.',
+  },
+  lastKnownLocation: {
+    id: 'LocationFieldset.lastKnownLocation',
+    defaultMessage: 'Last known location',
+    description: ('A label on a form field for the last known location of a '
+        + 'person.'),
+  },
+  lastKnownLocationInstructions: {
+    id: 'LocationFieldset.lastKnownLocationInstructions',
+    defaultMessage: ('Type an address or open the map to indicate the location '
+        + 'by clicking on the map.'),
+    description: ('Instructions for how to fill in a field indicating the last '
+        + 'known location of a person.'),
+  },
+  showMap: {
+    id: 'LocationFieldset.showMap',
+    defaultMessage: 'Show map',
+    description: 'Label on a button to show a map.',
+  },
+});
+
+const BASE_MAPS_API_URL = 'https://maps.googleapis.com/maps/api/js';
+
+class LocationFieldset extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showMap: false,
+      haveStartedLoadingMapScript: false,
+      haveFinishedLoadingMapScript: false,
+    };
+    this.mapsApiDisabled = ENV.maps_api_key == null || ENV.maps_api_key == '';
```

Comment:
Empty string is falsy in JS, so if you wanted you could just do `this.mapsApiDisabled = !!ENV.maps_api_key` (not sure if this is considered good style or not).

Also, I find it easier to understand to have "positive"-valued boolean variables -- so in this case IMO it would be better to have the variable be mapsApi**En**abled.